### PR TITLE
#72 Add the list of users

### LINKED ISSUE #72: [Feature] - Add the list of users
### What feature would you like to see?

**Please note this should only be worked on after https://github.com/jdwilkin4/fcc-forum-clone/issues/71 is completed**

By now, we should have all we need to create the list of users.
In the previous issue, we wrote a function `getUserAvatarComponent` which either returns an empty string(when user not found) or the component as a string.

### Task 1
Add new `div` element to the HTML which will contain the list of users.

### Task 2
Write `displayUsers` function in which you will iterate through all the users we obtained from the FCC Forum API, create the avatar component using `getUserAvatarComponent` for each of them and display them inside the div container you created in Task 1.

### Task 3
Call the `displayUsers` function in a place where it makes the most sense.

You can include some basic styling if you like but it's not necessary at this point.

### PR DESCRIPTION
**Pull request by JesusMurguia**
# Description

Added the list of users, included the shrinking/expanding transitions as well.

Fixes #72 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore items (this includes basic clean up of files, or updates to packages)
- [ ] Updates to documentation

## Checklist

- [X] I have a descriptive title for my PR
- [X] I have linked the issue to this [PR](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [X] I have run the project locally and reviewed the code changes
- [X] My changes generate no new warnings

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -137,6 +140,49 @@ function displayCategories() {
...
+  let onlineUsersAvatars = `<span>Online (${ids.length}):</span>`;
+  ids.forEach((id) => {
+    onlineUsersAvatars += getUserAvatarComponent(id);
+  });
+
+  userListContainer.innerHTML = onlineUsersAvatars;
+}
+
+//handles the collapsing/expanding transitions
+function activeCollapsableUserList() {
```
Comment (Dorfieeee) [inline index.js:156]: Hey,

great job again, I can see you have some experience writing code. 😸 

Couldn't we just replace this function with CSS property `transition-delay: 1s;` instead?

Either way, it's good to be merged. If you think the CSS property is better solution, you can change it. If not, could you provide an explanation why not?

Thank you

**Review by Dorfieeee — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (JesusMurguia) [reply index.js:155]: You are right! I didnt think about that, i replaced the function, thank you for the feedback!

File: index.js [hunk 2]
```diff
@@ -152,35 +152,13 @@ function displayUsers() {
...
-  function handleCollapse() {
-    mouseLeaveTimeout = setTimeout(
-      () => userListContainer.classList.add("collapsed"),
-      1000
-    );
-    //clear the mouseover timeout as well as the reference to skip the animation
-    clearTimeout(mouseOverTimeout);
-    mouseOverTimeout = null;
-  }
+  userListContainer.addEventListener("mouseover", () =>
```
Comment (Dorfieeee) [inline index.js:156]: I guess we could replace this with `:hover` as well.

**Review by Dorfieeee — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (JesusMurguia) [reply index.js]: done!

**Review by Dorfieeee — CHANGES_REQUESTED** [review]
Good enough.

**Review by Dorfieeee — APPROVED** [review]
Good enough.

**Review by AndreyDmitriev39r — APPROVED** [review]
LGTM

**Pull request merged**